sim/priest/priest.go

```diff
@@ -107,6 +109,9 @@ func New(char core.Character, selfBuffs SelfBuffs, talents proto.PriestTalents)
 	})
 
 	priest.registerShadowfiendCD()
+	if priest.Talents.PowerInfusion {
```

Comment:
Nit: would be slightly cleaner to put this check inside the registerPowerInfusionCD() function. Also would be consistent with implementation of other CDs.